### PR TITLE
Resolve 3.13 issues for `ctypes._endian`

### PR DESCRIPTION
**Pull request by max-muoto**
Starting in 3.13, we're no long doing a star import from `ctypes`: https://github.com/python/cpython/pull/105768

### CODE REVIEW

File: stdlib/ctypes/_endian.pyi [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import sys
-from _ctypes import RTLD_GLOBAL as RTLD_GLOBAL, RTLD_LOCAL as RTLD_LOCAL, Structure, Union
-from ctypes import DEFAULT_MODE as DEFAULT_MODE, cdll as cdll, pydll as pydll, pythonapi as pythonapi
 
-if sys.version_info >= (3, 12):
+if sys.version_info >= (3, 13):
+    from ctypes import Array as Array, Structure, Union
```
Comment (srittau) [inline stdlib/ctypes/_endian.pyi:4]: Why add `Array` as new re-export?

Comment (max-muoto) [reply stdlib/ctypes/_endian.pyi:4]: It's exported in the implementation: https://github.com/python/cpython/blob/094375b9b7e087a4f0f60541dc7f2dc53be92646/Lib/ctypes/_endian.py#L2

Comment (JelleZijlstra) [reply stdlib/ctypes/_endian.pyi:4]: It's just imported there. This is a private module, and it doesn't seem like it's intentionally re-exported.

Comment (max-muoto) [reply stdlib/ctypes/_endian.pyi:4]: I see, was trying to match the runtime as closely as possible, but makes sense.

Comment (max-muoto) [reply stdlib/ctypes/_endian.pyi]: https://github.com/python/typeshed/pull/12372/commits/2074c3da25d04aec81d9029760959c0f7b57b717

File: stdlib/ctypes/_endian.pyi [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import sys
-from _ctypes import RTLD_GLOBAL as RTLD_GLOBAL, RTLD_LOCAL as RTLD_LOCAL, Structure, Union
-from ctypes import DEFAULT_MODE as DEFAULT_MODE, cdll as cdll, pydll as pydll, pythonapi as pythonapi
+from _ctypes import Structure, Union
 
-if sys.version_info >= (3, 12):
+if sys.version_info < (3, 13):
+    from _ctypes import RTLD_GLOBAL as RTLD_GLOBAL, RTLD_LOCAL as RTLD_LOCAL
```
Comment (JelleZijlstra) [inline stdlib/ctypes/_endian.pyi:5]: All of these look like incidental re-exports, can we remove all the `as` imports?

Looks like it used to do `from ctypes import *`.

Comment (max-muoto) [reply stdlib/ctypes/_endian.pyi]: https://github.com/python/typeshed/pull/12372/commits/adb8d1059194c829cae6ea9a1ee03bec85746639

Comment (max-muoto) [reply stdlib/ctypes/_endian.pyi]: Let me also add an ignore for older Python versions.

**Comment by max-muoto** [comment]
@JelleZijlstra Should be ready for another look, apologies for the delay.

**Pull request merged**